nibabel/processing.py

```diff
@@ -310,3 +311,96 @@ def smooth_image(img,
                                    mode=mode,
                                    cval=cval)
     return out_class(sm_data, img.affine, img.header)
+
+
+def _transform_range(x, new_min, new_max):
+    """ Transform data to a new range, while maintaining ratios.
+
+    Parameters
+    ----------
+    x : array-like
+        The data to transform.
+    new_min, new_max : scalar
+        The minimum and maximum of the output array.
+
+    Returns
+    -------
+    transformed : array-like
+        A copy of the transformed data.
+
+    Examples
+    --------
+    >>> _transform_range([2, 4, 6], -1, 1)
+    array([-1.,  0.,  1.])
+    """
+    x = np.asarray(x)
+    x_min, x_max = x.min(), x.max()
+    return (x - x_min) * (new_max - new_min) / (x_max - x_min) + new_min
```

Comment:
should it become resilient to `x_max == x_min` when it would either set it to `new_min`  or blow some better than ZeroDivisonError exception?

Reply:
thank you @yarikoptic - i will account for `x_max == x_min` and i will add a test for this case.